### PR TITLE
PYTHON-1787: fix NotMasterError wrong attribute error

### PR DESCRIPTION
**Pull request by juliusgeo**
This is fixing the issue from the previous pull request which was caused by NotMasterError failing

https://jira.mongodb.org/browse/PYTHON-1787

### CODE REVIEW

File: pymongo/errors.py [hunk 1]
```diff
@@ -114,7 +114,7 @@ class NotMasterError(AutoReconnect):
     Subclass of :exc:`~pymongo.errors.AutoReconnect`.
     """
     def __str__(self):
```
Comment (ShaneHarvey) [inline pymongo/errors.py:116]: As discussed, can you add a test that asserts that the "full error" string is included in str(exc) and [traceback.format_exc()](https://docs.python.org/3/library/traceback.html#traceback.format_exc?

File: test/test_monitoring.py [hunk 2]
```diff
@@ -439,6 +439,9 @@ def test_not_master_error(self):
             client.pymongo_test.test.find_one_and_delete({})
         except NotMasterError as exc:
             error = exc.errors
+            import traceback
+            self.assertIn("full error", str(exc))
+            self.assertIn("full error", traceback.format_exc())
```
Comment (ShaneHarvey) [inline test/test_monitoring.py:444]: This is exactly what I had in mind. What do you think about moving these assertions to their own test cases (maybe to a new file `test_errors.py`)? You can use the error classes directly, like:
```
exc = NotMasterError(...)
self.assertIn("full error", str(exc))
try:
    raise exc
except NotMasterError:
    self.assertIn("full error", traceback.format_exc())
```

File: test/test_errors.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+                  unittest)
+
+class TestErrors(PyMongoTestCase):
+    def test_not_master_error(self):
+        exc = NotMasterError("not master test", "details")
+        self.assertIn("full error", str(exc))
+        try:
+            raise exc
+        except NotMasterError:
+            import traceback
```
Comment (ShaneHarvey) [inline test/test_errors.py:17]: Can you move this import to the top of the file?

File: test/test_errors.py [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+                            OperationFailure)
+from test import (client_context,
+                  client_knobs,
+                  PyMongoTestCase,
+                  sanitize_cmd,
+                  unittest)
+
+class TestErrors(PyMongoTestCase):
+    def test_not_master_error(self):
+        exc = NotMasterError("not master test", "details")
```
Comment (ShaneHarvey) [inline test/test_errors.py:12]: Suggest using something like `{ok: 0, errmsg:"error"}` instead of "details".

File: test/test_errors.py [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+                            NotMasterError,
+                            OperationFailure)
+from test import (client_context,
+                  client_knobs,
+                  PyMongoTestCase,
+                  sanitize_cmd,
+                  unittest)
+
+class TestErrors(PyMongoTestCase):
+    def test_not_master_error(self):
```
Comment (ShaneHarvey) [inline test/test_errors.py:11]: Can you also add a similar test for OperationFailure?

File: test/test_errors.py [hunk 6]
```diff
@@ -0,0 +1,18 @@
+from pymongo.errors import (AutoReconnect,
```
Comment (ShaneHarvey) [inline test/test_errors.py:1]: Add the copyright notice `# Copyright 2020-present MongoDB, Inc. ....`

Comment (ShaneHarvey) [inline test/test_errors.py:1]: After importing standard python libraries but before importing pymongo can you add `sys.path[0:0] = [""]`. For example: https://github.com/mongodb/mongo-python-driver/blob/4760d0781505c0f13d5d1bc061ff89075927eb2f/test/test_bson.py#L27

File: test/test_errors.py [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+
+class TestErrors(PyMongoTestCase):
+    def test_not_master_error(self):
+        exc = NotMasterError("not master test", "details")
+        self.assertIn("full error", str(exc))
+        try:
+            raise exc
+        except NotMasterError:
+            import traceback
+            self.assertIn("full error", traceback.format_exc())
```
Comment (ShaneHarvey) [inline test/test_errors.py:18]: Can you add the `if __name__ == "__main__":` logic at the bottom of this file? https://github.com/mongodb/mongo-python-driver/blob/4760d0781505c0f13d5d1bc061ff89075927eb2f/test/test_bson.py#L1088-L1089

File: test/test_errors.py [hunk 8]
```diff
@@ -0,0 +1,18 @@
+from pymongo.errors import (AutoReconnect,
+                            NotMasterError,
+                            OperationFailure)
+from test import (client_context,
+                  client_knobs,
+                  PyMongoTestCase,
+                  sanitize_cmd,
+                  unittest)
```
Comment (ShaneHarvey) [inline test/test_errors.py:8]: Suggest removing any unused imports.

File: pymongo/errors.py (hunk 1 above)
Comment (juliusgeo) [reply pymongo/errors.py:116]: Done.

File: test/test_errors.py (hunk 8 above)
Comment (juliusgeo) [reply test/test_errors.py:20]: Done.

File: test/test_errors.py (hunk 5 above)
Comment (juliusgeo) [reply test/test_errors.py:24]: Done.

File: test/test_errors.py (hunk 7 above)
Comment (juliusgeo) [reply test/test_errors.py:30]: Done.

**Pull request merged**